catalog/resource_catalog.go

```diff
@@ -69,7 +67,7 @@ func ResourceCatalog() *schema.Resource {
 				return fmt.Errorf("cannot remove new catalog default schema: %w", err)
 			}
 			d.SetId(ci.Name)
-			return nil
+			return updateOwner(ctx, d, c)
```

Comment:
```suggestion
			return updateFunctionFactory("/unity-catalog/catalogs/", []string{"owner"})(ctx, d, c)
```

don't introduce single-use variables that make readability harder :)